### ISSUE TITLE
Eraser is broken

### ISSUE THREAD

**Issue by yurivict** [xournalpp/xournalpp#1339]
It shows as a gigantic square, no control over its size. It erases vert roughly, together with a lot of things around it.

Version 1.0.12

**Comment by LittleHuba** [comment]
Please attach your settings file `<home>/.xournalpp/settings.xml`. You may also try to set the eraser size in the menubar to `small`.
Also attach a picture of the bug please. Otherwise it is hard to exactly pin what goes wrong.

**Comment by yurivict** [comment]
https://people.freebsd.org/~yuri/xournal-settings.xml

**Comment by yurivict** [comment]
https://people.freebsd.org/~yuri/xournal-screenshot-2019-07-11_02.24.00.mp4

**Comment by LittleHuba** [comment]
This could be related to #858...

**Comment by yurivict** [comment]
It seems that it should:
1. Have a "brush size" control in pixels
2. Replace a square eraser with appropriately sized eraser that looks like an eraser

Please look how gimp does erasing, it is pretty good there.

**Comment by lehmanju** [comment]
@yurivict you can actually change the size of the eraser by using the same buttons to change stroke width when eraser tool is activated

**Comment by yurivict** [comment]
> you can actually change the size of the eraser by using the same buttons to change stroke width when eraser tool is activated

The same button doesn't change the size. The size can be changed with the pen size buttons to the right (Thin/Medium/Thick). But the size of the eraser doesn't reflect the size of the pen, and the eraser doesn't resemble the eraser, it's just a square of the same size, regardless of the size.

**Comment by lehmanju** [comment]
True, pen size and eraser size are independent, however, I don't think one should combine them. 

> and the eraser doesn't resemble the eraser, it's just a square of the same size, regardless of the size.

What exactly do you mean by that?

**Comment by yurivict** [comment]
> What exactly do you mean by that?

https://people.freebsd.org/~yuri/gimp-eraser-in-action-2019-07-22_23.15.20.mp4

**Comment by lehmanju** [comment]
So you would like to have more control over it like it Gimp but this then would rather be an enhancement instead of a bug, at least I can't see how the current behavior is not intended.

**Comment by yurivict** [comment]
Yes, it's an enhancement.

**Comment by LittleHuba** [comment]
Smooth erasers are impossible for us. We use vectors for the strokes and the support for smooth edged eraser is absolutely over the top. We would have to actually redraw the stroke pixel by pixel, interpolate the actual curves and change the color depending on the distance to the eraser...

We can implement circle eraser and such things. The icon should then reflect the eraser form as it does currently for the rectangular one.

**Comment by Izerpizer** [comment]
A cool addition to this would be to allow the eraser size to change based on how hard you are pressing (if in that mode) with the stylus, in the exact same way that pen size changes based on how hard you press.

**Comment by KronosTheLate** [comment]
I think I have posted about this elsewhere, but it seems relevant to mention it here:
The whiteout eraser has a rectangular indicator, but erases as a circle. The problem is demonstrated in [this](https://drive.google.com/file/d/1MuPayVmaA16a4zSu9VIKXZIrzkvpZYbT/view?usp=sharing) video.

**Comment by MMachado05** [comment]
@LittleHuba looking at what @KronosTheLate posted, could the eraser at least be changed to better represent the "shape" of what's going to be erased? The extra corners that won't actually erase anything feel very unproductive.

**Comment by WildPenquin** [comment]
I was wondering why I can not reproduce this... just in case someone is wondering the same thing, this only applies to eraser in "whiteout" mode, "standard" works. But they are a different thing indeed. and whiteout does not indeed work as the GUI would indicate.
